src/compile/mark/mark.ts

```diff
@@ -308,7 +308,8 @@ export function getSort(model: UnitModel): VgCompare {
 function getMarkGroup(model: UnitModel, opt: {fromPrefix: string} = {fromPrefix: ''}) {
   const {mark, markDef, encoding, config} = model;
 
-  const clip = getFirstDefined(markDef.clip, scaleClip(model), projectionClip(model));
+  const clipDef = getFirstDefined(markDef.clip, scaleClip(model), projectionClip(model));
+  const clip = isSignalRef(clipDef) ? clipDef : clipDef ? true : undefined;
```

Comment:
nit: Do we really need this line?

is clipDef isn't a signal ref, it's basically a boolean.  `clipDef ? true : undefined` seems a bit unnecessary?  (esp. given that `scaleClip` and `projectionClip` return `true | undefined` already.